addons/account/models/sequence_mixin.py

```diff
@@ -50,6 +50,20 @@ def init(self):
     def _must_check_constrains_date_sequence(self):
         return True
 
+    def _sequence_matches_date(self):
+        self.ensure_one()
+        date = fields.Date.to_date(self[self._sequence_date_field])
+        sequence = self[self._sequence_field]
+
+        if not sequence or not date:
+            return False
+
+        format_values = self._get_sequence_format_param(sequence)[1]
+        year_match = (format_values['year']
+                      and format_values['year'] == date.year % 10**len(str(format_values['year'])))
+        month_match = format_values['month'] and format_values['month'] == date.month
```

Comment:
If there is no year in the sequence, then the year matches (because there is no constraint on it). Same for the month.